### PR TITLE
small bugfixes

### PR DESCRIPTION
**Pull request by TrevorBoone**
two small bugs I found while working on the end to end tests.

### CODE REVIEW

File: external/fetcher.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func DownloadStream(r io.ReadCloser, checksum, localPath string) error {
 	if err != nil {
 		return err
 	}
-	file, err := os.Create(localPath)
+	file, err := os.OpenFile(localPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0700)
```
Comment (TrevorBoone) [inline external/fetcher.go]: Create doesn't by default give execute permissions (even to owner)

File: policies/recipes/artifacts.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func fetchArtifact(ctx context.Context, artifact *osconfigpb.SoftwareRecipe_Arti
 		extension = path.Ext(uri.Path)
 		checksum = remote.Checksum
 		cl := &http.Client{}
-		reader, err := getHTTPArtifact(cl, *uri)
```
Comment (TrevorBoone) [inline policies/recipes/artifacts.go]: reader shadowed previous variable

File: external/fetcher.go (hunk 1 above)
Comment (adjackura) [reply external/fetcher.go]: We should pass in the permissions or leave it up to the caller to set them after, 0700 does not seem like a normal permission set for a function called simply called DownloadStream

Comment (TrevorBoone) [reply external/fetcher.go]: so after some thought I added a chmod in the execution step for artifacts, but not local files. I figured changing permissions on local files was not something we wanted to do in this api.

**Pull request merged**